src/powershell/Program.cs

```diff
@@ -433,7 +445,7 @@ private static void ThrowOnFailure(string call, int code)
             CallingConvention = CallingConvention.Cdecl,
             CharSet = CharSet.Ansi,
             SetLastError = true)]
-        private static extern int Exec(string path, string[] args);
+        private static extern int Exec(string path, string?[] args);
```

Comment:
Why args could be null?

Reply:
Not `args` but its elements - `string?[]` vs ` string[]?`. In line 330 - `execArgs[execArgs.Length - 1] = null;`.

Reply:
That null is required by `exec` to indicate the end of args I believe, and is probably not going to pass to pwsh. Did you verify that the null will actually be passed to `Main` in case of a shell login?

Reply:
Main() on Unix never gets args as null because Unic convention is - args[0] always point to program name.

I see no reason why we could not pass a null argument among others - `string?[] args` says about this.

As side notice, I don't found a documented convention for C# Main(). Will ask in .Net Runtime repo (https://github.com/dotnet/runtime/issues/40449).

Reply:
My comment was wrong, I intended to ask about `Start([MarshalAs(...)]string?[] args` in `ManagedEntrance.cs`. Will leave another comment.